### DISCUSSION TITLE
Where to put custom framework extensions?

### DISCUSSION THREAD

**Discussion by notramo** [martenframework/marten#212] [Q&A]
Similarly to #209, the documentation does not mention framework extensions.

Where to put these in the directory structure?
- context producers
- custom middleware
- template filters
- template tags
- schema fields
- etc.

The Marten project uses `ext/` for extending Marten classes, but these are not extensions of existing objects, but entirely new objects (which inherit from Marten classes).

**Comment by ellmetha** [comment]
Hey!  👋 

The response will be relatively similar. It's not expected that most projects will have to implement custom abstractions like these so at the moment projects can choose to define those wherever they want. 🙂 

For example:

* `context_producers` app folder for context producers
* `middlewares` app folder for middlewares
* `template_filters` app folder for template tags
* `template_tags` app folder for template tags
* etc

Ultimately we could add some guidelines regarding "preferred" folder structure and associated namings to the documentation.

**Reply by notramo** [reply]
I would recommend that as conventions help to increase productivity when scaling to bigger teams.
